### PR TITLE
Use GPU versions in components.json for CSE

### PR DESCRIPTION
**Pull request by ganeshkumarashok**
**What type of PR is this?**
/kind refactor
/kind design

**What this PR does / why we need it**:

This is to make CSE consume the same GPU driver versions from components.json (for Ubuntu). This will make it consistent with how the VHD now consumes it, and we will have a single source of truth for GPU drivers on Ubuntu.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: pkg/agent/datamodel/const.go [hunk 1]
```diff
@@ -132,19 +140,71 @@ const (
...
+	return nil
+}
+
+//nolint:gochecknoinits
+func init() {
+	if err := LoadConfig(); err != nil {
+		panic(fmt.Sprintf("Failed to load configuration: %v", err))
+	}
+}
+
```
Comment (lilypan26) [inline pkg/agent/datamodel/const.go:207]: maybe move all this to another file `gpu_components.go`? `consts.go` probably not the best place

Comment (ganeshkumarashok) [reply pkg/agent/datamodel/const.go]: fair, moved it

File: pkg/agent/datamodel/gpu_components_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	}
+
+	if AKSGPUCudaVersionSuffix == "" {
+		t.Error("NvidiaCudaDriverVersion is empty")
+	}
+
+	if AKSGPUGridVersionSuffix == "" {
+		t.Error(("AKSGPUGridVersionSuffix is empty"))
+	}
+}
```
Comment (lilypan26) [inline pkg/agent/datamodel/gpu_components_test.go:24]: can we add some tests for checking that the driver version and suffix is parsed correctly from components.json?

Comment (ganeshkumarashok) [reply pkg/agent/datamodel/gpu_components_test.go:51]: Added regex matching for parsed image driver version and suffix to ensure it's parsed properly.

**Review by lilypan26 — APPROVED** [review]

**Pull request merged**